w3-filecoin.md

```diff
@@ -0,0 +1,187 @@
+
+Storage Provider (web3.storage) must be given capabilities by a Filecoin broker Authority (Spade) to invoke capabilities to get its stored content into Filecoin deals.
+
+```mermaid
+sequenceDiagram
+    participant Authority as 🌐<br/><br/>did:web:spade.storage
+    participant Storage Provider as 🌐<br/><br/>did:web:web3.storage #32;
+
+    Authority->>Storage Provider: delegate `filecoin/add`
+```
+
+## Filecoin deals
+
+When a storage provider has enough 
+
+```mermaid
+sequenceDiagram
+    participant Authority as 🌐<br/><br/>did:web:spade.storage
+    participant Storage Provider as 🌐<br/><br/>did:web:web3.storage #32;
+
+    Storage Provider->>Authority: invoke `filecoin/add`
+    Note left of Authority: Prepare for deal
+    Authority-->>Storage Provider: receipt
+    Note left of Authority: Land deal in Filecoin
+    Authority->>Storage Provider: invoke `filecoin/deal`
+    Authority-->>Storage Provider: invoke `filecoin/fail`
+```
+
+## Capabilities
+
+This section describes the capabilities that form the w3 Filecoin protocol, along with details relevant to invoking capabilities with a service provider.
+
+### `filecoin/add`
+
+A Filecoin broker Authority MAY delegate capabilities to any storage provider.
+
+> `did:web:spade.storage` delegates `filecoin/add` capabilities to `did:web:web3.storage`
+
+```json
+{
+  "iss": "did:web:spade.storage",
+  "aud": "did:web:web3.storage",
+  "att": [{ "with": "did:web:spade.storage", "can": "filecoin/add" }],
+  "exp": null,
+  "sig": "..."
+}
+```
+
+> `did:web:web3.storage` invokes capability from `did:web:spade.storage`
+
+```json
+{
+  "iss": "did:web:web3.storage",
+  "aud": "did:web:spade.storage",
+  "att": [{
+    "with": "did:web:spade.storage",
+    "can": "filecoin/add",
+    "nb": : {
+      "aggregateId": "id...",
+      "cars": [
+        {
+          "link": "bag...",
+          "size": 1234,
+          "md5": "md5...",
+          "commP": "commP...",
+          "url": "https://..."
+        }
+      ]
+    }
+  }],
+  "prof": [
+    "iss": "did:web:spade.storage",
+    "aud": "did:web:web3.storage",
+    "att": [{ "with": "did:web:spade.storage", "can": "filecoin/add" }],
+    "exp": null,
+    "sig": "..."
+  ],
+  "sig": "..."
+}
+```
+
+### `filecoin/deal`
+
+A Filecoin broker authority MAY invoke deal capability when one aggregate lands into a Filecoin deal. It should include the CID of the request CAR cid from `filecoin/add` invocation.
+
+```json
+{
+  "iss": "did:web:spade.storage",
+  "aud": "did:web:spade.storage",
+  "att": [{
+    "with": "did:web:spade.storage",
+    "can": "filecoin/deal",
+    "nb": : {
+      "aggregateId": "id...",
+      "requestCarCid": "bag...",
+      "commP": "commP..."
+    }
+  }],
+  "prof": [],
+  "sig": "..."
+}
+```
+
+### `filecoin/fail`
+
+A Filecoin broker authority MAY invoke deal capability when one aggregate fails to land into a Filecoin deal. It should include the CID of the request CAR cid from `filecoin/add` invocation.
```

Comment:
I assume there are no partial failures per aggregate. If so I think having one invocation with `result` type telling if it's success or failure is a better design

Reply:
Yes, everything fails. We get a list of chunks that made it fail